lib/instance-validator.js

```diff
@@ -112,7 +112,7 @@ class InstanceValidator {
           .then(newError => { throw newError || error; }))
```

Comment:
Though it's maybe not clear from the context, this `then()` call is applied to the result of `runHooks('validationFailed')`. I don't know under what conditions `runHooks` can resolve with a value that is an error, but it semantically did not seem relevant to this pull request.